### PR TITLE
Built in/succeed smoketest

### PR DESCRIPTION
**Pull request by ededals**
Creates a built-in/succeed smoketest for #1240

### CODE REVIEW

File: infrastructure/tooling/src/smoketest/checks/succeed.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+      if (status.state === 'pending'){
+        await new Promise(resolve => setTimeout(resolve, 2000));
+      } else {
+        return {
+          ['built-in/succeed']: result.state,
+        };
+
+      }
+    }
+    return result;
```
Comment (djmitche) [inline infrastructure/tooling/src/smoketest/checks/succeed.js]: There's a lint error here due to the unreachable return.  When you've got the next revision ready to push, run `yarn lint` to find any details like that.

File: infrastructure/tooling/src/smoketest/checks/succeed.js [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+      metadata: {
+        name: "Smoketest built-in/succeed",
+        description: "built-in/succeed task created during smoketest",
+        owner: "smoketest@taskcluster.net",
+        source: "https://taskcluster.net",
+      },
+      payload: {},
+    };
+    let taskId = taskcluster.slugid();
+    let queue = new taskcluster.Queue({
```
Comment (djmitche) [inline infrastructure/tooling/src/smoketest/checks/succeed.js]: you can use `taskcluster.Queue(taskcluster.fromEnvVars())` as a shorter way to write the same thing.

File: infrastructure/tooling/src/smoketest/checks/succeed.js [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+      rootUrl: process.env.TASKCLUSTER_ROOT_URL,
+      credentials: {
+        clientId: process.env.TASKCLUSTER_CLIENT_ID,
+        accessToken: process.env.TASKCLUSTER_ACCESS_TOKEN,
+      },
+    });
+    let result = await queue.createTask(taskId, task);
+    while (1){
+      let status = await queue.status(taskId);
```
Comment (djmitche) [inline infrastructure/tooling/src/smoketest/checks/succeed.js]: The return value of this call has the shape `{status: { state: .. }}`, so you'd need to look at `status.status.state`, not `status.state`.

File: infrastructure/tooling/src/smoketest/checks/succeed.js [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+      if (status.state === 'pending'){
```
Comment (djmitche) [inline infrastructure/tooling/src/smoketest/checks/succeed.js]: ..so `status.state` is always undefined, so this is never true..

File: infrastructure/tooling/src/smoketest/checks/succeed.js [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+        await new Promise(resolve => setTimeout(resolve, 2000));
+      } else {
+        return {
```
Comment (djmitche) [inline infrastructure/tooling/src/smoketest/checks/succeed.js]: It'd be great to assert the state is `completed` here.

**Comment by ededals** [comment]
Hi, thanks for the comments :) I hope I managed to fix everything. 
One thing I am not sure about is, if using ```utils.skip(...)``` for the failing tests was the right choice.

File: infrastructure/tooling/src/smoketest/checks/succeed.js [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+      payload: {},
+    };
+    let taskId = taskcluster.slugid();
+    utils.status({message: 'built-in/succeed taskId: ' + taskId});
+    let queue = new taskcluster.Queue(taskcluster.fromEnvVars());
+    await queue.createTask(taskId, task);
+    let pollForStatus = true;
+    setTimeout(function () {
+      pollForStatus = false;
+    }, 12000);
```
Comment (djmitche) [inline infrastructure/tooling/src/smoketest/checks/succeed.js]: Needs one more `0` (this is only 12 seconds) :)

File: infrastructure/tooling/src/smoketest/checks/succeed.js [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+        return utils.skip({
+          [`built-in/succeed`]: status.status.state,
+          reason: 'Task finished with status ' + status.status.state,
+        });
+      }
+    }
+    return utils.skip({
+      [`built-in/succeed`]: 'deadline exceeded',
+      reason: 'Deadline exceeded',
+    });
```
Comment (djmitche) [inline infrastructure/tooling/src/smoketest/checks/succeed.js]: Good question.  In both of these cases, I think we can just throw an exception.

**Comment by ededals** [comment]
Yes this seems to bet working better and is also more concise.

**Pull request merged**